designs/operator-monitoring-plugin.md

```diff
@@ -0,0 +1,209 @@
+---
+title: controller-observation-plugin
+authors:
+  - "@Kavinjsir"
+reviewers:
+  - "@varsha"
+  - "@camila"
+approvers:
+creation-date: 2022-06-06
+last-updated: 2022-06-06
+status: provisional
+see-also:
+  - ""
+replaces:
+  - ""
+superseded-by:
+  - ""
+---
+
+# a-grafana-dashboard-plugin-to-observe-controller-metrics
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions [optional]
+
+- TBD
```

Comment:
WDYT about : 
a) Should we scaffold a dir with the config to install the grafana? See: https://grafana.com/docs/grafana/latest/installation/kubernetes/

b) What should be the name of the default directory and where it should be scaffold?
It seems that we could scaffold: (root dir) something like:

grafana/
grafana/dashboards
grafana/config

OR 

inside of the config dir another grafana dir with the k8s manifests to be applied on the cluster when the plugin is used

Reply:
We can also add a few other pointers like:
1. How is the plugin backwards compatible for users who have prometheus scaffolded?
- Is it compatible only with specific versions of prometheus? Or do can we have an existing kb project and add this plugin to scaffold grafana manifests smoothly. Do we need any other changes to an old project structure.

Reply:
That's great! I like all these questions, they helps exploring the project.  Thx @camilamacedo86 @varshaprasad96